### PR TITLE
add sticky headers and optional height

### PR DESCRIPTION
**Pull request by wattsjs**
made some changes locally to fit my needs - thought it may be useful 🙂

- add the ability to set specific items as sticky
- change `height` to optional, defaulting to 100%

### CODE REVIEW

**Comment by wattsjs** [comment]
run prettier

File: src/lib/list/List.svelte [hunk 1]
```diff
@@ -32,9 +32,11 @@
 
 	export let itemCount: number;
 	export let itemSize: number;
-	export let height: number;
+	export let height: number | string = "100%";
 	export let width = "100%";
 
+	export let stickyIndicies: number[] = [];
```
Comment (ghostebony) [inline src/lib/list/List.svelte:38]: fix the typo `stickyIndicies` to `stickyIndices`

File: src/lib/list/List.svelte [hunk 2]
```diff
@@ -171,6 +173,22 @@
 		style:height={isVertical ? `${innerSize}px` : "100%"}
 		style:width={!isVertical ? `${innerSize}px` : "100%"}
 	>
+		{#if stickyIndicies.length}
```
Comment (ghostebony) [inline src/lib/list/List.svelte:176]: check `indexes` too:
```svelte
{#if stickyIndices.length && indexes.length}
```

File: src/lib/list/List.svelte [hunk 3]
```diff
@@ -171,6 +173,22 @@
...
+			{@const stickyIndex = Math.max(
+				...stickyIndicies.filter((i) => i < (indexes?.[0] ?? 0))
```
Comment (ghostebony) [inline src/lib/list/List.svelte:178]:
```ts
...stickyIndices.filter((i) => i < indexes[0])
```

File: src/lib/list/List.svelte [hunk 4]
```diff
@@ -171,6 +173,22 @@
...
+			)}
+			{#if stickyIndex >= 0}
+				<div
+					style:position="sticky"
+					style:top={isVertical ? `${marginTop}px` : "0px"}
+					style:left={isVertical ? "0px" : `${marginLeft}px`}
+					style:z-index="1"
+				>
+					<slot name="item" index={stickyIndex} />
```
Comment (ghostebony) [inline src/lib/list/List.svelte:187]: add default template message like
```svelte
<slot name="item" {index}>Missing template</slot>
```

**Review by ghostebony — CHANGES_REQUESTED** [review]
thanks for the contribution.
after these changes I think it will be ready to merge

**Comment by wattsjs** [comment]
Sounds great 👍

**Review by wattsjs — COMMENTED** [review]
lgtm

**Review by ghostebony — APPROVED** [review]

**Pull request merged**